src/domain/bitcoin/index.types.d.ts

```diff
@@ -1,5 +1,7 @@
 type Satoshis = number & { readonly brand: unique symbol }
 type TargetConfirmations = number & { readonly brand: unique symbol }
 type MilliSatoshis = number & { readonly brand: unique symbol }
+type WholeBitcoin = number & { readonly brand: unique symbol }
```

Comment:
Don't quite understand this `WholeBitcoin` implies an `int` for me but I think its actually a float isn't it? I would prefer not to add this type and stick with a consistent `Sat` type and if translation is needed to talk to an external dependency do the translation in the service / repository implementation.
Or are there any other restrictions on this type other than where the decimal point is compared to `Satoshis`?

Reply:
yes this is a float. the idea is to convey that this number is denominated in bitcoin, not satoshis. 

the lightning wallet either takes millisats or sats as a unit of account. 
the bitcoind wallet takes bitcoin which is why I'm introducing this new type to make it more clear what number is being passed around. 

happy to have better suggestion for the name instead of whole bitcoin.

I think what you're saying also is that wholeBitcoin should be only present in the bitcoind internal service implementation instead of the the domain?

Reply:
Yes that is correct - in the domain we should be consistent and only have 1 type to represent bitcoin denominated value. I think `Satoshis` is the most natural here. That bitcoind wants a float is an implementation detail that should be taken care of in the service layer that talks to bitcoind.